sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorClientBuilder.java

```diff
@@ -698,22 +745,50 @@ public EventProcessorClientBuilder processPartitionClose(Consumer<CloseContext>
      * amount of additional network bandwidth consumption that is generally a favorable trade-off when considered
      * against periodically making requests for partition properties using the Event Hub client.</p>
      *
-     * @param trackLastEnqueuedEventProperties {@code true} if the resulting events will keep track of the last enqueued
-     * information for that partition; {@code false} otherwise.
+     * @param trackLastEnqueuedEventProperties {@code true} if the resulting events will keep track of the last
+     *     enqueued information for that partition; {@code false} otherwise.
+     *
      * @return The updated {@link EventProcessorClientBuilder} instance.
      */
     public EventProcessorClientBuilder trackLastEnqueuedEventProperties(boolean trackLastEnqueuedEventProperties) {
         this.trackLastEnqueuedEventProperties = trackLastEnqueuedEventProperties;
         return this;
     }
 
+    /**
+     * Sets the default starting position for each partition if a checkpoint for that partition does not exist in the
+     * {@link CheckpointStore}.
+     *
+     * <p>
+     * The starting position for each partition is determined using the following steps.
+     * <ol>
+     *     <li>Check the {@link CheckpointStore} for that partition's checkpoint.</li>
+     *     <li>If it does not exist, check {@link #initialPartitionEventPosition(Map)} for an entry.</li>
+     *     <li>If an entry does not exist, check if {@link #initialPartitionEventPosition(EventPosition)} is set.</li>
+     *     <li>Else, use {@link EventPosition#latest()}.</li>
+     * </ol>
+     *
+     * @param initialEventPosition Event position to start reading events.  Applies to every partition in the Event Hub
+     * without a checkpoint nor an entry in the {@link #initialPartitionEventPosition(Map) initial position map}.
+     *
+     * @return The updated {@link EventProcessorClientBuilder} instance.
+     */
+    public EventProcessorClientBuilder initialPartitionEventPosition(EventPosition initialEventPosition) {
```

Comment:
It seems the position set with this method applies to all partitions, so would it make sense to call it `initialEventPosition` to reflect this?

Reply:
I wasn't sure if it would be more discoverable to use overloads since we have this builder pattern.

Reply:
because I'm hoping to add another one that's a function which takes a partitionId and returns an EventPosition.

Reply:
I was also wondering if _initialEventPosition_ is better. But after looking at the new APIView and the "existing" GA-ed method for setting event position, I feel like defining this new method as an overload is more discoverable.

Reply:
yeah, I agree, it's more discoverable with the rest of the overloads and the messaging about only one of the overloads can be used will also be simpler.